fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/communication/AdHocModule.java

```diff
@@ -63,38 +57,37 @@ public void enable() {
      */
     @Override
     public void enable(AdHocModuleConfiguration configuration) {
+        super.enable(configuration);
 
         if (configuration == null) {
-            logEnableConfigurationNull();
-        } else {
-            this.configuration = configuration;
-
-            AdHocConfiguration.Builder builder = new AdHocConfiguration.Builder(owner.getId());
-
-            for (int i = 0; i < min(2, configuration.getNrOfRadios()); i++) {
-                AdHocModuleConfiguration.AdHocModuleRadioConfiguration radio = configuration.getRadios().get(i);
-                InterfaceConfiguration.Builder interfaceBuilder =
-                        new InterfaceConfiguration.Builder(radio.getChannel0())
-                                .ip(address)
-                                .subnet(subnet)
-                                .radius(radio.getDistance())
-                                .power(radio.getPower());
-                if (radio.getNrOfChannels() > 1) {
-                    interfaceBuilder.secondChannel(radio.getChannel1());
-                }
-                builder.addInterface(interfaceBuilder.create());
-            }
+            return;
+        }
 
-            owner.sendInteractionToRti(new AdHocCommunicationConfiguration(owner.getSimulationTime(), builder.create()));
+        AdHocConfiguration.Builder builder = new AdHocConfiguration.Builder(owner.getId());
+        for (int i = 0; i < min(2, configuration.getNrOfRadios()); i++) {
```

Comment:
Any reasons why we don't use a for each loop here? Would also allow us to get rid of the somewhat unnecessary method `getNrOfRadios()`